Sources/Flush.swift

```diff
@@ -86,53 +89,47 @@ class Flush: AppLifecycle {
     }
 
     func flushQueueInBatches(_ queue: Queue, type: FlushType) {
-        var mutableQueue = queue
-        while !mutableQueue.isEmpty {
-            var shouldContinue = false
-            let batchSize = min(mutableQueue.count, APIConstants.batchSize)
-            let range = 0..<batchSize
-            let batch = Array(mutableQueue[range])
-            let ids: [Int32] = batch.map { entity in
-                (entity["id"] as? Int32) ?? 0
-            }
-            // Log data payload sent
-            Logger.debug(message: "Sending batch of data")
-            Logger.debug(message: batch as Any)
-            let requestData = JSONHandler.encodeAPIData(batch)
-            if let requestData = requestData {
-                let semaphore = DispatchSemaphore(value: 0)
-                #if os(iOS)
+        flushBatchQueue.sync {
```

Comment:
put all calls to `flushQueueInBatches` into sync queue in case it's being called from multiple threads